zyte_spider_templates/spiders/job_posting.py

```diff
@@ -0,0 +1,247 @@
+from pydantic import BaseModel, ConfigDict, Field
+from scrapy import Request
+from scrapy.crawler import Crawler
+from scrapy_poet import DummyResponse
+from scrapy_spider_metadata import Args
+from zyte_common_items import JobPosting, JobPostingNavigation, ProbabilityRequest
+
+from zyte_spider_templates.spiders.base import (
+    ARG_SETTING_PRIORITY,
+    INPUT_GROUP,
+    BaseSpider,
+)
+from zyte_spider_templates.utils import get_domain
+
+from ..documentation import document_enum
+from ..params import (
+    CustomAttrsInputParam,
+    CustomAttrsMethodParam,
+    ExtractFromParam,
+    GeolocationParam,
+    MaxRequestsParam,
+    UrlParam,
+    UrlsFileParam,
+    UrlsParam,
+)
+from ..utils import load_url_list
+
+if TYPE_CHECKING:
+    # typing.Self requires Python 3.11
+    from typing_extensions import Self
+
+
+@document_enum
+class JobPostingCrawlStrategy(str, Enum):
+    category: str = "category"
+    """Follow pagination and job posting detail pages."""
+
+    direct_item: str = "direct_item"
+    """Treat input URLs as direct links to job posting detail pages, and extract a
+    job posting from each."""
+
+
+class JobPostingCrawlStrategyParam(BaseModel):
+    crawl_strategy: JobPostingCrawlStrategy = Field(
+        title="Crawl strategy",
+        description="Determines how input URLs and follow-up URLs are crawled.",
+        default=JobPostingCrawlStrategy.category,
+        json_schema_extra={
+            "enumMeta": {
+                JobPostingCrawlStrategy.category: {
+                    "title": "Category",
+                    "description": "Follow pagination and job posting detail pages.",
+                },
+                JobPostingCrawlStrategy.direct_item: {
+                    "title": "Direct URLs to job postings",
+                    "description": (
+                        "Treat input URLs as direct links to job posting detail pages, and "
+                        "extract a job posting from each."
+                    ),
+                },
+            },
+        },
+    )
+
+
+class JobPostingSpiderParams(
+    CustomAttrsMethodParam,
+    CustomAttrsInputParam,
+    ExtractFromParam,
+    MaxRequestsParam,
+    GeolocationParam,
+    JobPostingCrawlStrategyParam,
+    UrlsFileParam,
+    UrlsParam,
+    UrlParam,
+    BaseModel,
+):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "groups": [
+                INPUT_GROUP,
+            ],
+        },
+    )
+
+
+class JobPostingSpider(Args[JobPostingSpiderParams], BaseSpider):
+    """Yield job postings from a job website.
+
+    See :class:`~zyte_spider_templates.spiders.job_posting.JobPostingSpiderParams`
+    for supported parameters.
+
+    .. seealso:: :ref:`job-posting`.
+    """
+
+    name = "job_posting"
+
+    metadata: Dict[str, Any] = {
+        **BaseSpider.metadata,
+        "title": "Job posting",
+        "description": "Template for spiders that extract job posting data from job websites.",
```

Comment:
```suggestion
        "description": "Template for spiders that extract job posting data from websites.",
```

zyte.com is not a job website, but you can extract job posting data from its jobs page.